Assets/BossRoom/Scripts/Server/Game/State/ServerCharSelectState.cs

```diff
@@ -169,39 +170,55 @@ public override void NetworkStart()
             }
             else
             {
-                NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnected;
                 NetworkManager.Singleton.OnClientDisconnectCallback += OnClientDisconnectCallback;
                 CharSelectData.OnClientChangedSeat += OnClientChangedSeat;
 
-                if (IsHost)
-                {
-                    // host doesn't get an OnClientConnected()
-                    // and other clients could be connects from last game
-                    // So look for any existing connections to do intiial setup
-                    var clients = NetworkManager.Singleton.ConnectedClientsList;
-                    foreach (var net_cl in clients)
-                    {
-                        OnClientConnected(net_cl.ClientId);
-                    }
-                }
+                // Cosmin: Question? Should OnNotifyServerClientLoadedScene delegate also cover
+                // When a new client joins in? If yes, then this we will become obsolete for our case here
+                NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnected;
+                // NSS: Suggested Fix, use the OnNotifyServerClientLoadedScene to let you know that the client
+                // is initialized, all scene NetworkObjects have been loaded, and can be considered "ready to send and receive messages".
+                NetworkManager.Singleton.SceneManager.OnNotifyServerClientLoadedScene += SceneManager_OnNotifyServerClientLoadedScene;
```

Comment:
```suggestion
                NetworkManager.Singleton.OnClientConnectedCallback += (clientId) => { SeatNewPlayer(clientId); }; // this isn't tested, but would help make this part a bit simpler?
                // NSS: Suggested Fix, use the OnNotifyServerClientLoadedScene to let you know that the client
                // is initialized, all scene NetworkObjects have been loaded, and can be considered "ready to send and receive messages".
                NetworkManager.Singleton.SceneManager.OnNotifyServerClientLoadedScene += (_, clientId) => { SeatNewPlayer(clientId); }; // same here
```

Reply:
You can just do `+= SeatNewPlayer;` directly on line 178. It's necessary for line 181 though.

Reply:
I still believe having a OnPlayerConnected function is somewhat better since it is more explicit and clear to the user in what it does, + if we ever need to add additional functionality when a new player connects that is the best place to do it without having to modify the class in all different places